src/tensor_product_layer.jl

```diff
@@ -0,0 +1,29 @@
+abstract type TensorProducLayer <: Function end
+
+struct TPLayer{Array{TensorProductBasis},S<:AbstractArray,b<:AbstractArray,Int} <: TensorProducLayer
```

Comment:
```suggestion
struct TPLayer{Array{TensorProductBasis},S<:AbstractArray,b<:AbstractArray,Int} <: TensorProductLayer
```

Reply:
Maybe it would be more informative to rename `TensorProductLayer ` -> `AbstractTensorProductLayer` and `TPLayer ` -> `TensorProductLayer`